### PR TITLE
Fixing Bug:1136164. Case insensitive interpolation of apprelease blobs.

### PR DESCRIPTION
**Pull request by tyagi-iiitv**

### CODE REVIEW

**Comment by rail** [comment]
I wonder what the performance penalty is of using regular expressions instead of plain `replace()`...

**Comment by tyagi-iiitv** [comment]
It'll be difficult to test all possible cases without using regular expressions. Is there any way we can check the performance degrade using regular expressions in this case?

**Comment by tyagi-iiitv** [comment]
The strings to compare are small in this case, so maybe it shouldn't affect the performance that much.

File: auslib/test/blobs/test_apprelease.py [hunk 1]
```diff
@@ -669,17 +669,17 @@ def setUp(self):
...
     "openURL": "http://example.org/url/%LOCALE%",
-    "notificationURL": "http://example.org/notification/%LOCALE%",
-    "alertURL": "http://example.org/alert/%LOCALE%",
+    "notificationURL": "http://example.org/notification/%LOCaLE%",
+    "alertURL": "http://example.org/alert/%LocaLE%",
     "showPrompt": false,
     "showNeverForVersion": true,
     "fileUrls": {
-        "c1": "http://a.com/%FILENAME%"
+        "c1": "http://a.com/%filename%"
```
Comment (mozbhearsum) [inline auslib/test/blobs/test_apprelease.py]: Can you update this to ensure that %OS_FTP%, %PRODUCT%, %OS_BOUNCER%, %VERSION%, and %OS% are tested, too? Have a look at one of the Firefox releases in your local dev environment for some examples of URLs that contain them.

**Comment by tyagi-iiitv** [comment]
I've made changes accordingly, please verify.

**Comment by tyagi-iiitv** [comment]
Done adding tests. This PR can be reviewed for merging now.

File: auslib/test/blobs/test_apprelease.py [hunk 2]
```diff
@@ -865,13 +865,15 @@ def testSchema2WithOptionalAttributes(self):
         returned = self.blobK.getInnerXML(updateQuery, "minor", self.whitelistedDomains, self.specialForceHosts)
         returned_footer = self.blobK.getInnerFooterXML(updateQuery, "minor", self.whitelistedDomains, self.specialForceHosts)
         returned = [x.strip() for x in returned]
+        print "I'm here"
+        print str(returned)
```
Comment (mozbhearsum) [inline auslib/test/blobs/test_apprelease.py]: Please remove these :)

Comment (tyagi-iiitv) [inline auslib/test/blobs/test_apprelease.py]: Oops, this went unnoticed .. !!

**Pull request merged**